searx/engines/docker_hub.py

```diff
@@ -0,0 +1,69 @@
+"""
+Docker Hub (IT)
+
+@website     https://hub.docker.com
+@provide-api yes (https://docs.docker.com/registry/spec/api/)
+
+@using-api   yes
+@results     JSON
+@stable      yes
+@parse       url, title, content, thumbnail, publishedDate
+"""
+
+from json import loads
+from dateutil import parser
+from urllib.parse import urlencode
+
+categories = ['it']  # optional
+paging = True
+
+base_url = "https://hub.docker.com/"
+search_url = base_url + "api/content/v1/products/search?{query}&type=image&page_size=25"
+
+
+def request(query, params):
+    '''pre-request callback
+    params<dict>:
+      method  : POST/GET
+      headers : {}
+      data    : {} # if method == POST
+      url     : ''
+      category: 'search category'
+      pageno  : 1 # number of the requested page
+    '''
```

Comment:
I guess these doc-strings can be dropped .. if there is something special to this engine, it is good to write down a doc-string, but if it is only C&P better leaf it.